main.go

```diff
@@ -63,7 +63,16 @@ func newAppForUser(dataFileName string) todoApp {
 	if err != nil {
 		panic(fmt.Sprintf("could not get current user: %v", err))
 	}
-	return todoApp{DataFilePath: fmt.Sprintf("%s/%s", usr.HomeDir, dataFileName)}
+	dataPath := fmt.Sprintf("%s/%s", usr.HomeDir, dataFileName)
```

Comment:
I'm not sure that this is the right place to handle the existence of the file. Even if you stat the file and it's not there, it might be there by the time you want to access it (however remote that possibility). My thought is that we can handle it pretty simply by using 
```go
f, err := os.OpenFile(app.DataFilePath, os.O_CREATE|os.O_RDONLY, 0644)
```
within the `read` function where we are currently using the Open fuction.  What do you think of that?

Reply:
I guess that's a totally valid approach, sure!

Reply:
do you want to make that change?

Reply:
I can, I'm not able to do this right now though. Feel free to go ahead with it, otherwise I'll try it in the upcoming days.

Reply:
Ok, I will make the change in this branch. Thanks for the collaboration!

Reply:
oops - just realized - I'm used to working where everyone has access to the same repo. I will close in favor of a new PR.